src/output_writers.jl

```diff
@@ -403,6 +413,43 @@ mutable struct Checkpointer{I, T, P, A} <: AbstractOutputWriter
            verbose :: Bool
 end
 
+"""
+    Checkpointer(model; frequency=nothing, interval=nothing, dir=".", prefix="checkpoint",
+                        force=false, verbose=false,
+                        properties = [:architecture, :boundary_conditions, :grid, :clock,
+                                      :eos, :constants, :closure, :velocities, :tracers,
+                                      :timestepper])
+
+Construct a `Checkpointer` that checkpoints the model to a JLD2 file every so often as
+specified by `frequency` or `interval`. The `model.clock.iteration` is included in the
+filename to distinguish between multiple checkpoint files.
+
+Note that extra model `properties` can be safely specified, but removing crucial properties
+such as `:velocities` will make restoring from the checkpoint impossible.
```

Comment:
Maybe open an issue to ensure that `velocities`, `tracers`, and `timestepper` are checkpointed (in the same way that `grid` is ensured)...